### PR TITLE
Implement Annual Client Report

### LINKED ISSUE #636: (design) Client Report
This issue concerns the design of the client report.  It should look like the mockup below.

client report
_Fig 1: Client Report_

The data from this report should only come from the `period_total` table.  The SQL statement will look up the debtor group accounts and retrieve the opening balances from the period 0 of the `period_total` table.  It will then sum the `period_total` grouped by account_id where the fiscal year is 
the selected fiscal year.

As in 1.x, the client report could have two versions: a "detailed" version that shows debits/credits for opening balances and a "plain" version that only includes the combined balance.  The

### PR DESCRIPTION
**Pull request by jniles**
This commit completes the annual debtor group report started by @jeremielodi.  It provides a way to view a non-aged report for all debtors groups on a given fiscal year.

Closes #636.
Closes #2210.

**Screenshots**
clientsinreport
_Fig 1: Default Presentation of Annual Clients Report_

noclientsinreport
_Fig 2: Presentation if there are no clients_

### CODE REVIEW

**Comment by jniles** [comment]
Looks like I have to update a few tests before this should be reviewed.

**Comment by jniles** [comment]
Alright, this can be reviewed anytime now 👍

**Comment by jniles** [comment]
@jeremielodi, can I get a review?

File: client/src/modules/reports/generate/debtorBalanceReport/debtorBalanceReport.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+angular.module('bhima.controllers')
+  .controller('debtorBalanceReportController', DebtorBalanceReportController);
+
+DebtorBalanceReportController.$inject = [
+  '$state', '$sce', 'NotifyService', 'BaseReportService', 'AppCache',
+  'BaseReportService', 'reportData',
+];
+
```
Comment (jeremielodi) [inline client/src/modules/reports/generate/debtorBalanceReport/debtorBalanceReport.js:8]: @jniles, your code looks correct but there are less comments, comments are very important for helping others to easily understand what you do.

File: test/end-to-end/shared/components/bhFiscalYearSelect.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/* global  element, by */
+const FU = require('../FormUtils');
+
+const selector = '[bh-fiscal-year-select]';
+module.exports = {
+  set : (fiscalYear, id) => {
+    const locator = id ? by.id(id) : by.css(selector);
+    FU.select('$ctrl.fiscalId', fiscalYear, element(locator));
+  },
+};
```
Comment (jeremielodi) [inline test/end-to-end/shared/components/bhFiscalYearSelect.js:10]: perfect!

**Comment by jniles** [comment]
@jeremielodi, thank you for your review!  I'll add some comments and then schedule a merge with bors.

**Pull request merged**